packages/ember/addon/index.ts

```diff
@@ -0,0 +1,57 @@
+import * as Sentry from '@sentry/browser';
+import { addGlobalEventProcessor, SDK_VERSION, BrowserOptions } from '@sentry/browser';
+import environmentConfig from 'ember-get-config';
+
+import { next } from '@ember/runloop';
+import { assert, warn, runInDebug } from '@ember/debug';
+import Ember from 'ember';
+
+export function SentryForEmber(_runtimeConfig: BrowserOptions | undefined) {
+  const config = environmentConfig['@sentry/ember'];
+  assert('Missing configuration', config);
+  assert('Missing configuration for Sentry.', config.sentry);
+
+  const initConfig = Object.assign({}, config.sentry, _runtimeConfig || {});
+
+  createEmberEventProcessor();
+
+  Sentry.init(initConfig);
+
+  runInDebug(() => {
+    if (config.ignoreEmberOnErrorWarning) {
+      return;
+    }
+    next(null, function () {
+      warn(
+        'Ember.onerror found. Using Ember.onerror can hide some errors (such as flushed runloop errors) from Sentry. Use Sentry.captureException to capture errors within Ember.onError or remove it to have errors caught by Sentry directly. This error can be silenced via addon configuration.',
```

Comment:
Interesting. So we ask users to instrument `Ember.onError` theirselves? Any reason we didn't opt for automatic instrumentation of `Ember.onError`?

Reply:
Not quite, `Ember.onerror` doesn't fully cover all error, just some of the errors within Ember. If someone doesn't have `Ember.onerror` declared, then all errors will naturally forward to Sentry. If they **do** have onerror declared and are doing something with those errors, then a subset of errors might be getting ignored and not forwarded to Sentry, hence the warning.

Reply:
Oh, I misunderstood, that makes much more sense. Cool 👍